src/commands/help.ts

```diff
@@ -1,37 +1,63 @@
    *
    * @param message the message provided
    * Now automatically adds new commands - make sure the .js file has a prefix and description,
    */
-  command: async (message) => {
-    const helpEmbed: MessageEmbedOptions = {
-      color: '#0099FF',
-      title: 'Bot Information',
-      description:
-        'Hello! I am a test bot created by bradtaniguchi and members of ' +
-        'FreeCodeCamp.org to experiment with the process of building a ' +
-        'Discord Bot. You can view my source code at ' +
-        ' https://github.com/bradtaniguchi/discord-bot-test',
-      fields: [],
-      footer: { text: 'I am not affiliated with FreeCodeCamp in any way.' }
-    };
-    COMMANDS.forEach((commandDef) => {
-      helpEmbed.fields?.push({
-        name: commandDef.prefix,
-        value: commandDef.description.substring(0, 1023),
-        inline: false
-      });
-    });
-    await message.author.send({ embed: helpEmbed });
-    message.channel.send('Help message sent!');
+  command: async (message, { config }) => {
+    const param = message.content.split(' ')[2];
+    if (param == undefined) {
+      const helpEmbed: MessageEmbed = new MessageEmbed()
+        .setColor('#0099FF')
+        .setTitle('Help Command')
+        .addField(
+          'Bot Information',
+          'Hello! I am a test bot created by bradtaniguchi and members of ' +
+            'FreeCodeCamp.org to experiment with the process of building a ' +
+            'Discord Bot. You can view my [source code here](https://github.com/bradtaniguchi/discord-bot-test)'
+        )
+        // .addField(
+        //   'Commands',
+        //   `Hey there ${message.author}, here is the full list of all my commands.\n` +
+        //     `For more information about each command, use \`${config.PREFIX} help <command>\``
+        // )
+        .addField(
+          'List of Commands',
+          COMMANDS.sort((a, b) =>
+            a.prefix.toUpperCase() > b.prefix.toUpperCase() ? 1 : -1
+          )
+            .map((command, i) => {
+              if ((i + 1) % 4 === 0) {
+                return `\`${command.prefix}\n\``;
+              } else return `\`${command.prefix}\``;
+            })
+            .join(' ') +
+            '\n \n' +
+            `For more information about each command, use \`${config.PREFIX} help <command>\``
+        )
+        .setFooter('I am not affiliated with FreeCodeCamp in any way.');
+      message.channel.send(helpEmbed);
+    } else {
+      const commandFound = COMMANDS.find((command) => command.prefix == param);
+      if (!commandFound)
+        return message.channel.send(
+          `Cannot find the command you specified -> \`${param}\``
+        );
+      const helpEmbed = new MessageEmbed()
+        .setColor('#0099FF')
+        .setTitle(commandFound.usage)
+        .addField('Description', `\`${commandFound.description}\``)
+        .addField('\u200b', 'Required: `<>` | Optional: `[]`');
+      message.channel.send(helpEmbed);
```

Comment:
The bot should send messages to the user in a DM rather than in chat, to prevent spam.

Reply:
The original command sent the embed directly to the author, and a "Help message sent" confirmation to the channel (to prevent the bot appearing to be unresponsive to other users).